lib/src/back_end/code.dart

```diff
@@ -132,49 +149,230 @@ final class _MarkerCode extends Code {
+      case _NewlineCode():
+        // If formatting has been disabled, then don't write the formatted
+        // output. The unformatted output will be written when formatting is
+        // re-enabled.
+        if (_disableFormattingStart == -1) {
+          _buffer.write(_lineEnding);
+          if (code._blank) _buffer.write(_lineEnding);
+          _indent = code._indent;
+        }
+
+      case _TextCode():
+        // If formatting has been disabled, then don't write the formatted
+        // output. The unformatted output will be written when formatting is
+        // re-enabled.
+        if (_disableFormattingStart == -1) {
+          // Write any pending indentation.
+          _buffer.write(_indents[_indent] ?? (' ' * _indent));
+          _indent = 0;
+
+          _buffer.write(code._text);
+        }
+
+      case GroupCode():
+        _indent = code._indent;
+        for (var i = 0; i < code._children.length; i++) {
+          var child = code._children[i];
+          traverse(child);
+        }
+
+      case _MarkerCode():
+        if (_disableFormattingStart == -1) {
+          // Calculate the absolute offset from the beginning of the formatted
+          // output where the selection marker will appear based on how much
+          // formatted output we've written, pending indentation, and then the
+          // relative offset of the marker into the subsequent [Code] we will
+          // write.
+          var absolutePosition = _buffer.length + _indent + code._offset;
+          switch (code._marker) {
+            case _Marker.start:
+              _selectionStart = absolutePosition;
+            case _Marker.end:
+              _selectionEnd = absolutePosition;
+          }
+        } else {
+          // The marker appears inside a region where formatting is disabled.
+          // In that case, calculating where the marker will end up in the
+          // final formatted output is more complicated because we haven't
+          // actually written any of the code between the `// dart format off`
+          // comment and this marker to [_buffer] yet. However, we do know the
+          // *absolute* position of the selection markers in the original
+          // source.
+          //
+          // Let's say the source file looks like:
+          //
+          //               1         2         3
+          //     0123456789012345678901234567890123456789
+          //     bef  +  ore off code | inside on more
+          //
+          // Here, `bef  +  ore` is some amount of code appearing before
+          // formatting is disabled, `off` is the `// dart format off` comment,
+          // `code` is some code inside the unformatted region, `|` is the
+          // selection marker, `inside` is more code in the unformatted region,
+          // `on` turns formatting back on, and `more` is formatted code at the
+          // end.
+          //
+          // We know the beginning of the unformatted region is at offset 10
```

Comment:
I don't understand how "just after the comment" is "offset 10". How are you counting offset?

Reply:
Oops! Sorry for the confusion. I tweaked the sample line of code after writing the paragraph explaining it and then forgot to go back and fix the offsets. Fixed.